### PR TITLE
Added support for Ocean ECS cluster Roll operations

### PR DESCRIPTION
**Pull request by janetlinkiruba**
Checklist
---------

JIRA Ticket -

Checklist (Please mark Yes/No/NA for each of the questions below) -

- I have provided clear and detailed explanation of changes made in the proposed PR : Yes
- I have compiled the code successfully my laptop : Yes
- I have written examples to demonstrates the usage of the newly implemented APIs : Yes
- I have incremented the release version : No
- I have run the Intellij Code Analysis on newly added code and fixed all (valid) reported issues : No

OR

- I have taken approval from scrum master or project manager for skipping any of the items above : No

### CODE REVIEW

File: src/main/java/com/spotinst/sdkjava/model/api/ocean/ecs/ApiGetEcsClusterNodesResponse.java [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+    private Integer             workloadRequestedCpuUnit;
+    private Integer         workloadRequestedMemoryInMiB;
+    private Integer                 workloadRequestedGpu;
+    private Integer             headroomRequestedCpuUnit;
+    private Integer         headroomRequestedMemoryInMiB;
+    private Integer                 headroomRequestedGpu;
+    private Integer                    registeredCpuUnit;
+    private Integer                registeredMemoryInMiB;
+    private Integer                        registeredGpu;
+    private String                    registrationStatus;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/ocean/ecs/ApiGetEcsClusterNodesResponse.java:29]: Should it be enumeration?

File: src/main/java/com/spotinst/sdkjava/model/api/ocean/ecs/ApiGetEcsClusterNodesResponse.java [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+import java.util.HashSet;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiGetEcsClusterNodesResponse implements IPartialUpdateEntity {
+    @JsonIgnore
+    private Set<String>                            isSet;
+    private String                             lifeCycle;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/ocean/ecs/ApiGetEcsClusterNodesResponse.java:18]: Should it be enumeration?

File: src/main/java/com/spotinst/sdkjava/model/api/ocean/ecs/ApiGetEcsClusterNodesResponse.java [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+    private String			                    publicIp;
+    private Integer             workloadRequestedCpuUnit;
+    private Integer         workloadRequestedMemoryInMiB;
+    private Integer                 workloadRequestedGpu;
+    private Integer             headroomRequestedCpuUnit;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/ocean/ecs/ApiGetEcsClusterNodesResponse.java:23]: Should be double?

File: src/main/java/com/spotinst/sdkjava/model/api/ocean/ecs/ApiImportClusterComputeConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+import com.spotinst.sdkjava.client.rest.IPartialUpdateEntity;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiImportClusterComputeConfiguration implements IPartialUpdateEntity {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/ocean/ecs/ApiImportClusterComputeConfiguration.java:16]: Rename to ApiImportClusterResponseComputeConfiguration

File: src/main/java/com/spotinst/sdkjava/model/api/ocean/ecs/ApiImportClusterComputeConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    @JsonIgnore
+    private Set<String>                         isSet;
+    private List<String>                        subnetIds;
+    private ApiImportClusterLaunchSpecification launchSpecification;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/ocean/ecs/ApiImportClusterComputeConfiguration.java:20]: Rename to ApiImportClusterResponseLaunchSpecification

File: src/main/java/com/spotinst/sdkjava/model/api/ocean/ecs/ApiImportClusterLaunchSpecInstanceMetadataOptions.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiImportClusterLaunchSpecInstanceMetadataOptions implements IPartialUpdateEntity {
+    @JsonIgnore
+    private Set<String> isSet;
+    private Integer     httpPutResponseHopLimit;
+    private String      httpTokens;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/ocean/ecs/ApiImportClusterLaunchSpecInstanceMetadataOptions.java:19]: Should it be enum.

File: src/main/java/com/spotinst/sdkjava/model/api/ocean/ecs/ApiImportClusterLaunchSpecInstanceMetadataOptions.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    private String      httpEndpoint;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/ocean/ecs/ApiImportClusterLaunchSpecInstanceMetadataOptions.java:20]: Should it be enum.

File: src/main/java/com/spotinst/sdkjava/model/bl/ocean/ecs/GetEcsClusterNodesResponse.java [hunk 8]
```diff
@@ -0,0 +1,420 @@
...
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class GetEcsClusterNodesResponse {
+    @JsonIgnore
+    private Set<String>                            isSet;
+    private String                             lifeCycle;
+    private String			                    publicIp;
+    private Integer             workloadRequestedCpuUnit;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/bl/ocean/ecs/GetEcsClusterNodesResponse.java:13]: should be double.

**Review by anuragsharma-123 — CHANGES_REQUESTED** [review]
Please check review comments and incorporate.

File: src/main/java/com/spotinst/sdkjava/model/api/ocean/ecs/ApiGetEcsClusterNodesResponse.java (hunk 2 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/model/api/ocean/ecs/ApiGetEcsClusterNodesResponse.java:18]: yes updated

File: src/main/java/com/spotinst/sdkjava/model/api/ocean/ecs/ApiImportClusterComputeConfiguration.java (hunk 5 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/model/api/ocean/ecs/ApiImportClusterComputeConfiguration.java:20]: fixed

File: src/main/java/com/spotinst/sdkjava/model/bl/ocean/ecs/GetEcsClusterNodesResponse.java (hunk 8 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/model/bl/ocean/ecs/GetEcsClusterNodesResponse.java:13]: fixed

File: src/main/java/com/spotinst/sdkjava/model/api/ocean/ecs/ApiImportClusterLaunchSpecInstanceMetadataOptions.java (hunk 7 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/model/api/ocean/ecs/ApiImportClusterLaunchSpecInstanceMetadataOptions.java:20]: fixed

**Review by anuragsharma-123 — APPROVED** [review]
Reviewed fixes. Approved.

**Pull request merged**